### ISSUE TITLE
Next 13 beta RSC fetch not working with public directory

### ISSUE THREAD

**Issue by TomasHubelbauer** [vercel/next.js#42760]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.1.0: Sun Oct  9 20:14:54 PDT 2022; root:xnu-8792.41.9~2/RELEASE_X86_64
    Binaries:
      Node: 19.0.0
      npm: 8.19.2
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.0.3-canary.3
      eslint-config-next: 13.0.2
      react: 18.2.0
      react-dom: 18.2.0

### How are you deploying your application? (if relevant)

next export

### Describe the Bug

I am using Next 13 beta with the app API (no pages routes) and I am hoping to build a component which pulls text from a TXT file in the public directory into its children. Here's the code:

```javascript
import Link from 'next/link';

export default async function RSCTest() {
  const text = await fetch('/rsc-test.txt').then(response => response.text());
  return (
    <>
      <header>
        <Link href="">« Home</Link>
      </header>
      {text}
    </>
  );
}
```

Unfortunately I am getting this error:

> Error: Failed to parse URL from /rsc-test.txt

![rsc-error](https://user-images.githubusercontent.com/6831144/201205508-6d126980-e4b8-4689-9ffe-5a37da5ca7cb.png)

### Expected Behavior

I expect the text to be loaded using the fetch, additionally, I expect `next export` to bake the result of that fetch into the component at build time.

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code will be addressed faster

https://github.com/TomasHubelbauer/next-13-ssg

### To Reproduce

```
npm install
npm run dev
```

After, access http://localhost:300 and click the first route link (the one with RSC in the name).

**Comment by shuding** [comment]
`/rsc-test.txt` is a relative path, you can't fetch that on the server side (in a server components, or a client component during SSR). You should use `fs` to import that file directly, or use an absolute URL.

**Comment by TomasHubelbauer** [comment]
Thanks for the tip @shuding!
I used Node's `fs` to read the file and now the route works as expected!
One thing worth noting is that the FS path is relative to the project root, so I had to read `public/rsc-test.txt`.
Just in case anyone comes across this issue and gets stuck on this.

However, even after applying this fix, the issue with `node export` crashing that's also happening in #42762 started happening here.
Since I already opened that ticket, I'll leave this one closed and hopefully the root of the problem will be diagnosed in the linked issue and fix both this and that issue.

I updated the reproduction repository with your suggested  fix.
If you want to reproduce this, remove the RCC routes and try running `next build && next export` with just the RSC route so that the problems with the RCC routes don't impact the reproducibility of this issue.
